ui-admin/src/study/surveys/CreateSurveyModal.tsx

```diff
@@ -116,3 +87,37 @@ const CreateSurveyModal = ({ studyEnvContext, isReadOnlyEnv, show, setShow }: {s
 }
 
 export default CreateSurveyModal
+
+/** hook for a form that sets a name and stableId */
+export const useFormCreationNameFields = () => {
```

Comment:
I don't have a strong opinion about it.  do you have a leaning?

Reply:
I mainly pointed it out because I've seen some hooks that have their own files, but others that don't. I lean towards having it in a separate file since it's sort of arbitrary that it lives in this particular Modal. Might also make it easier to know that these re-usable hooks exist

Reply:
cool, I'll keep it as-is in this PR, and then separate it out in the upcoming PR that uses it fro pre-reg surveys